jscomp/syntax/src/react_jsx_common.ml

```diff
@@ -63,3 +63,28 @@ let removeArity binding =
     | _ -> expr
   in
   {binding with pvb_expr = removeArityRecord binding.pvb_expr}
+
+let add_async_attribute ~async (body : Parsetree.expression) =
```

Comment:
Aren't these functions already in the file that deals with async transformations?

Reply:
If you mean the files that deal with async transformation are `ast_async.ml` or `ast_attributes.ml`, the `syntax` module doesn't have the `frontend` as dependency yet. That's why I added here again.
Do you want me to add the frontend module to deps?

Reply:
Oops, not able to do that due do circular deps: frontend -> common -> syntax

Reply:
Looks like front-end already accesses `ml`:

```lisp
(library
 (name frontend)
 (wrapped false)
 (flags
  (:standard -w +a-4-9-40-42-70))
 (libraries common ml))
```

and `ml` is where e.g. `ast_untagged_variants.ml` is.

Reply:
Maybe I'm missing something. `syntax` can access `ml`, but not `frontend` that has async transformation functions.

Reply:
Perhaps my wording is confusing. Actually, this is correct: frontend <- common <- syntax

Reply:
Put it where `ast_untagged_variants.ml` is, and change nothing else.

Reply:
It's not allowed to use anything that is not already in `ml`, obviously.

Reply:
Got it. https://github.com/rescript-lang/rescript-compiler/pull/6399/commits/8126a139ccb0e2458423aa298fda34aea2ee4460 
I was going to put only `ast_async.ml` in `ml`, but I thought it would be nice to have `ast_await.ml` in the same place, so I moved it. https://github.com/rescript-lang/rescript-compiler/pull/6399/commits/4d1e792d561a90e423157e95d18b3f053b003265